lndclient/invoices_client.go

```diff
@@ -111,10 +115,17 @@ func (s *invoicesClient) SubscribeSingleInvoice(ctx context.Context,
 				return
 			}
 
+			preimage, err := lntypes.MakePreimage(invoice.RPreimage)
```

Comment:
Doesn't this return an error when the preimage is still unknown?

Reply:
I guess we need to check for `ContractSettled` here?